src/_imagingft.c

```diff
@@ -614,11 +616,11 @@ font_getsize(FontObject* self, PyObject* args)
     /* calculate size and bearing for a given string */
 
     PyObject* string;
-    if (!PyArg_ParseTuple(args, "O|zOz:getsize", &string, &dir, &features, &lang)) {
+    if (!PyArg_ParseTuple(args, "O|izOz:getsize", &string, &mask, &dir, &features, &lang)) {
```

Comment:
Can we add the new parameter to the end, to maintain compatibility?

People have been calling `ImageFont.font.getsize` directly. A quick example:

```python
font = ImageFont.truetype(fontname, fontsize)
limits = font.getsize(text)
```
* https://github.com/brumar/SupperWrapper/blob/b8c58c9f8ac44bec25ca7397f58c568e270e264f/superWrapper.py#L123
* via https://github.com/search?l=Python&q=font.getsize+ImageFont&type=Code

Reply:
The linked and quoted example is the Python-layer function. It looks confusing as the code refers to the `FreeTypeFont` object as `font`, but updated function would have to be called with `font.font.getsize(...)`. The same is true for the search results in the first few pages.

There are three functions concerning the size of text:
```python
ImageDraw.Draw.textsize(...)
ImageFont.FreeTypeFont.getsize(...)  # the Python version; I did not update this
ImageFont.FreeTypeFont.font.getsize(...)  # the C version; updated by this PR
```

The equivalent code would be:
```python
font = ImageFont.truetype(...)
font.getsize(...)  # the Python function
font.font.getsize(...)  # the C function
```

I did not update the Python function as I think it would confusing to have the arguments in a different order, but not backwards compatible to add it at the front (suggestions are welcome).

If you still think the C function is a concern, would something with `PyArg_ParseTupleAndKeywords` be a good solution?

Reply:
Thanks, I think we're fine to change this then. The C API is internal and doesn't need to maintain backwards compatibility.